copy/copy.go

```diff
@@ -899,6 +899,12 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		return types.BlobInfo{}, errors.Wrap(err, "Error writing blob")
 	}
 
+	// If we can modify the layer's blob, set the desired compression for it to be set in the manifest.
+	if canModifyBlob && !isConfig {
+		uploadedInfo.Compression = c.dest.DesiredLayerCompression()
```

Comment:
Shouldn’t this be directly based on `compressionOperation` and/or the logic that set it instead of second-guessing the logic above here? It’s going to be difficult to keep the two in sync.

Reply:
It should be `compressionOperation`. Good catch!